testsuite/tests/properties/useless_bind/test.py

```diff
@@ -0,0 +1,52 @@
+"""
+Check that warnings are correctly issued when a dynamic variable is bound but
+is not used in the expression.
+"""
+
+from __future__ import absolute_import, division, print_function
+
+from langkit.dsl import ASTNode, LexicalEnv, LogicVar, T, UserField
+from langkit.expressions import (Bind, DynamicVariable, Property, Self, Var,
+                                 langkit_property, ignore)
+from langkit.parsers import Grammar, Or
+
+from utils import emit_and_print_errors
+
+dyn_node = DynamicVariable('dyn_node', T.FooNode)
+dyn_node_2 = DynamicVariable('dyn_node_2', T.FooNode)
+
+class FooNode(ASTNode):
+    @langkit_property(public=True)
```

Comment:
For testcase readability, I would suggest to rename the various properties depending on what they are used for. For instance: `test_propX` for the properties that constitute the warning test, and `helperX` for properties that exist only to provide material for tests.